sdks/java/core/src/main/java/org/apache/beam/sdk/coders/BigEndianShortCoder.java

```diff
@@ -62,16 +62,6 @@ public Short decode(InputStream inStream) throws IOException, CoderException {
   @Override
   public void verifyDeterministic() {}
 
-  /**
-   * {@inheritDoc}
-   *
-   * @return {@code true}. This coder is injective.
-   */
-  @Override
-  public boolean consistentWithEquals() {
```

Comment:
Does each of these need an implementation of structuralValue? Are there tests that fail if it is set to the wrong thing?

Reply:
As I understand all coders by default have base structuralValue from org.apache.beam.sdk.coders.Coder#structuralValue.
I think each of them doesn't need an implementation of structuralValue.
And we have some coders that override base behaviour of structuralValue.

I have found two tests which are broken (because they expect true result for consistentWithEquals)
org.apache.beam.sdk.coders.MapCoderTest > testConsistentWithEquals FAILED
org.apache.beam.sdk.coders.ListCoderTest > testConsistentWithEquals FAILED
by this reason I didn't remove overriding of consistentWithEquals from:
- org.apache.beam.sdk.coders.MapCoder
- org.apache.beam.sdk.coders.ListCoder

Reply:
The base structural value is the encoded bytes of the value. It is very expensive to compute. For any coder where `consistentWithEquals` returns `true`, that should be replaced with an override for `structuralValue` that returns the original value.

Tests of the exactly return value of `consistentWithEquals` are silly and can be deleted.

Reply:
I added `structuralValue` overriding and unit tests